crates/sui-bridge-indexer/src/eth_worker.rs

```diff
@@ -0,0 +1,158 @@
+            let eth_bridge_event = EthBridgeEvent::try_from_eth_log(log);
+            if eth_bridge_event.is_none() {
+                continue;
+            }
+            let bridge_event = eth_bridge_event.unwrap();
+            let block_number = log.block_number;
+            let block = provider.get_block(log.block_number).await.unwrap().unwrap();
+            let timestamp = block.timestamp.as_u64() * 1000;
+            let transaction = provider
+                .get_transaction(log.tx_hash)
+                .await
+                .unwrap()
+                .unwrap();
+            let gas = transaction.gas;
+            let tx_hash = log.tx_hash;
+
+            println!("Observed Finalized Eth bridge event: {:#?}", bridge_event);
+
+            match bridge_event {
+                EthBridgeEvent::EthSuiBridgeEvents(bridge_event) => match bridge_event {
+                    EthSuiBridgeEvents::TokensDepositedFilter(bridge_event) => {
+                        println!("Observed Finalized Eth Deposit");
+                        let transfer = TokenTransfer {
+                            chain_id: bridge_event.source_chain_id,
+                            nonce: bridge_event.nonce,
+                            block_height: block_number,
+                            timestamp_ms: timestamp,
+                            txn_hash: tx_hash.as_bytes().to_vec(),
+                            txn_sender: bridge_event.sender_address.as_bytes().to_vec(),
+                            status: TokenTransferStatus::Deposited,
+                            gas_usage: gas.as_u64() as i64,
+                            data_source: BridgeDataSource::Eth,
+                            data: Some(TokenTransferData {
+                                sender_address: bridge_event.sender_address.as_bytes().to_vec(),
+                                destination_chain: bridge_event.destination_chain_id,
+                                recipient_address: bridge_event.recipient_address.to_vec(),
+                                token_id: bridge_event.token_id,
+                                amount: bridge_event.sui_adjusted_amount,
+                            }),
+                        };
+
+                        let _ = write(pool, vec![transfer]);
+                    }
+                    EthSuiBridgeEvents::TokensClaimedFilter(_)
+                    | EthSuiBridgeEvents::PausedFilter(_)
+                    | EthSuiBridgeEvents::UnpausedFilter(_)
+                    | EthSuiBridgeEvents::UpgradedFilter(_)
+                    | EthSuiBridgeEvents::InitializedFilter(_) => (),
+                },
+                EthBridgeEvent::EthBridgeCommitteeEvents(_)
+                | EthBridgeEvent::EthBridgeLimiterEvents(_)
+                | EthBridgeEvent::EthBridgeConfigEvents(_)
+                | EthBridgeEvent::EthCommitteeUpgradeableContractEvents(_) => (),
+            }
+        }
+    }
+}
+
+pub async fn process_unfinalized_eth_events(
```

Comment:
this function and `process_finalized_eth_events` are basically the same. Can we try to refactor it? some ideas:
1. the first 10+s line before the match statement is the same. Put them into a function
2. conversion from `TokensDepositedFilter` to `TokenTransfer` can be done by using a function
3. same for `TokensClaimedFilter` to `TokenTransfer`